packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -157,24 +159,26 @@ function nodeDataToMapTree(
 ): ExclusiveMapTree {
 	// A special cache path for processing unhydrated nodes.
 	// They already have the mapTree, so there is no need to recompute it.
-	const flexNode = tryGetInnerNode(data);
-	if (flexNode !== undefined) {
-		if (isMapTreeNode(flexNode)) {
-			if (!allowedTypes.has(getSimpleNodeSchemaFromNode(flexNode))) {
+	const innerNode = tryGetInnerNode(data);
+	if (innerNode !== undefined) {
+		if (isMapTreeNode(innerNode)) {
+			if (!allowedTypes.has(getSimpleNodeSchemaFromNode(innerNode))) {
 				throw new UsageError("Invalid schema for this context.");
 			}
 			// TODO: mapTreeFromNodeData modifies the trees it gets to add defaults.
 			// Using a cached value here can result in this tree having defaults applied to it more than once.
 			// This is unnecessary and inefficient, but should be a no-op if all calls provide the same context (which they might not).
 			// A cleaner design (avoiding this cast) might be to apply defaults eagerly if they don't need a context, and lazily (when hydrating) if they do.
 			// This could avoid having to mutate the map tree to apply defaults, removing the need for this cast.
-			return flexNode.mapTree as ExclusiveMapTree;
+			return innerNode.mapTree as ExclusiveMapTree;
 		} else {
 			// The node is already hydrated, meaning that it already got inserted into the tree previously
 			throw new UsageError("A node may not be inserted into the tree more than once");
 		}
 	}
 
+	assert(!isTreeNode(data), "data is known not to be a node after the above check.");
```

Comment:
It will never be hit, its just making the type checker happy, but since it apparently confused you (this has nothing to do with caching), I'll change the message.
```suggestion
	assert(!isTreeNode(data), "data without an inner node cannot be TreeNode");
```